### PR TITLE
Add matches/:id/join endpoint

### PR DESCRIPTION
**Pull request by andrewdt23**
#23

### CODE REVIEW

File: backend/app/models/match.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Match < ApplicationRecord
     validates :user1_id, allow_blank: false, :numericality => { :greater_than_or_equal_to => 0 }
-    validates :user2_id, :numericality => { :greater_than_or_equal_to => 0 }
+    validates :user2_id, allow_blank: true, :numericality => { :greater_than_or_equal_to => 0 }
```
Comment (scottpchow23) [inline backend/app/models/match.rb]: Are we sure we don't want `allow_nil: true` instead/as well?

Comment (scottpchow23) [reply backend/app/models/match.rb]: Also seems like we can make `user1_id` and `user2_id` foreign key constraints; any reason why we might not?

**Review by scottpchow23 — CHANGES_REQUESTED** [review]

File: backend/app/models/match.rb (hunk 1 above)
Comment (andrewdt23) [reply backend/app/models/match.rb]: I can add `allow_nil: true`. It seemed like it was working even when `user2_id` wasn't provided but it doesn't hurt.

Comment (andrewdt23) [reply backend/app/models/match.rb]: By foreign key constraints are you referring to adding `has_many :users` to the match model?

Comment (scottpchow23) [reply backend/app/models/match.rb]: Yeah something like that.

Comment (andrewdt23) [reply backend/app/models/match.rb]: I added the `allow_nil: true`. As far as the foreign key constraints, I double checked and they are already in place. When you try to create a match with a user id that does not exist, it violates the foreign key constraint and fails.

**Review by scottpchow23 — APPROVED** [review]
Looks good to me.

**Pull request merged**